### PR TITLE
Testsuit

### PR DESCRIPTION
**Pull request by sdiazpier**
This PR partially addresses issue #15. The testsuit must be further extended but this is a startpoint.

### CODE REVIEW

**Comment by alperyeg** [comment]
Output should be captured to a certain degree too.

File: l2l/tests/test_sa_optimizer.py [hunk 1]
```diff
@@ -1,69 +1,31 @@
 import unittest
 
+from .test_optimizer import OptimizerTestCase
```
Comment (alperyeg) [inline l2l/tests/test_sa_optimizer.py:3]: Can you refactor this to an absoulte path, i.e. `from l2l.tests.test_optimizer import OptimzerTestCase`

File: l2l/tests/test_optimizer.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+            bench_functs.get_function_by_index(function_id, noise=True)
+
+
+        self.experiment = Experiment(root_dir_path='/home/sandra/Documents/L2L/results')
+        jube_params = {}
+        self.trajectory, all_jube_params = self.experiment.prepare_experiment(name='L2L',
+                                                              log_stdout=True,
+                                                              jube_parameter = jube_params)
+        self.optimizee_parameters = namedtuple('OptimizeeParameters', [])
+        self.optimizee = FunctionGeneratorOptimizee(self.trajectory,benchmark_function, seed=1)
```
Comment (alperyeg) [inline l2l/tests/test_optimizer.py:26]: Last two lines are not pep8 confirm.

File: l2l/tests/test_gd_optimizer.py [hunk 3]
```diff
@@ -1,53 +1,41 @@
...
         try:
-            optimizer.post_process()
+
+            self.experiment.run_experiment(optimizee=self.optimizee,
+                                  optimizee_parameters=self.optimizee_parameters,
+                                  optimizer=optimizer,
+                                  optimizer_parameters=optimizer_parameters)
         except Exception:
-            self.fail()
+            self.fail(Exception.__name__)
```
Comment (alperyeg) [inline l2l/tests/test_gd_optimizer.py:32]: I would prefer a more pythonic way `except Exception as e ... `. Where does `self.fail` come from?

**Review by alperyeg — CHANGES_REQUESTED** [review]

**Comment by alperyeg** [comment]
There is also lot of history comments which are unrelated. I can rebase when merging later.

File: l2l/tests/test_sa_optimizer.py (hunk 1 above)
Comment (sdiazpier) [reply l2l/tests/test_sa_optimizer.py:3]: Yes sorry for that

File: l2l/tests/test_gd_optimizer.py (hunk 3 above)
Comment (sdiazpier) [reply l2l/tests/test_gd_optimizer.py:32]: It comes from unittest. I will change the format :)

**Review by alperyeg — APPROVED** [review]
Looks good to me.

**Comment by alperyeg** [comment]
Let me know if you want it merged.

**Pull request merged**